### DISCUSSION TITLE
Is there a way to maintain auth status in Webview?

### DISCUSSION THREAD

**Discussion by ericbaek** [pocketbase/pocketbase#4224] [Q&A]
I am currently building a React Native (Expo) app with Pocketbase as a backend.
We have certain part of the app implemented as a webview. So here is my question.

Is there a way to maintain user's authdata from the app to the webview?
I tried using exportToCookie() and passing it as a headers, but unsuccessful.
```
export default function index() {
    const cookie = pb.authStore.exportToCookie();
    console.log(cookie);

    return (
        <WebView
            source={{
                uri: 'URL',
                headers: {
                    Cookie: cookie
                },
            }}
        />
    )
}
```

Thanks.

**Comment by ganigeorgiev** [comment]
There is no builtin mechanism to do that.

You can export it as cookie (or other header) but then from inside the WebView you need to read that value and load it into the JS SDK instance within the WebView.

Another thing to consider if you want to keep the WebView `pb.authStore` synced with the one in your regular app, you'll need some way to notify the React Native for the changes in the WebView. This could be done by registering [`pb.authStore.onChange(func)`](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#common-auth-store-fields-and-methods) function that will submit a postMessage to the app and from where you'll call something like `pb.authStore.save(viewToken, viewModel)`.

I'm not familiar with the Expo WebView component so I'm not sure what is available but I would advice to search for a way how to send/receive messages from and to the WebView.

**Reply by ganigeorgiev** [reply]
_If `WebView` in the above code sample is from the `react-native-webview` package, you could check [Communicating between JS and Native](https://github.com/react-native-webview/react-native-webview/blob/master/docs/Guide.md#communicating-between-js-and-native)._
